magda-gateway/src/createBaseProxy.ts

```diff
@@ -24,6 +24,22 @@ export default function createBaseProxy(): httpProxy {
         ) {
             proxyRes.headers["Cache-Control"] = "public, max-age=60";
         }
+        /**
+         * Remove security sensitive headers
+         * `server` header is from scala APIs
+         * Proxied content has to be filtered from here
+         * while other content (produced locally by gateway) has been
+         * taken care of by `app.disable("x-powered-by");` in index.js
+         */
+        Object.keys(proxyRes.headers).forEach(headerKey => {
+            const headerKeyLowerCase = headerKey.toLowerCase();
+            if (
+                headerKeyLowerCase === "x-powered-by" ||
+                headerKeyLowerCase === "server"
+            ) {
+                delete proxyRes.headers[headerKey];
```

Comment:
Can we use `proxyRes.headers.headerKey = undefined` instead? `delete` has some performance issues: https://jsperf.com/delete-vs-undefined-vs-null/16